src/pages/history/History.jsx

```diff
@@ -0,0 +1,43 @@
+import { useState, useEffect } from "react";
+import { Link } from "react-router-dom";
+
+import { useDataContext } from "../../context/dataContext";
+import "./history.css";
+
+export default function History() {
+  const { state } = useDataContext();
+
+  const [history, setHistory] = useState([]);
+
+  const getHistoryData = () => state.history;
```

Comment:
This seems like a redundant callback. You can probably just fetch the history directly.